core/node/api_server/src/web3/tests/mod.rs

```diff
@@ -1102,3 +1112,114 @@ impl HttpTest for GenesisConfigTest {
 async fn tracing_genesis_config() {
     test_http_server(GenesisConfigTest).await;
 }
+
+#[derive(Debug)]
+struct FeeHistoryTest;
+
+#[async_trait]
+impl HttpTest for FeeHistoryTest {
+    async fn test(
+        &self,
+        client: &DynClient<L2>,
+        pool: &ConnectionPool<Core>,
+    ) -> anyhow::Result<()> {
+        let mut connection = pool.connection().await?;
+        let block1 = L2BlockHeader {
+            batch_fee_input: scaled_sensible_fee_input(1.0),
+            base_fee_per_gas: 100,
+            ..create_l2_block(1)
+        };
+        store_custom_l2_block(&mut connection, &block1, &[]).await?;
+        let block2 = L2BlockHeader {
+            batch_fee_input: scaled_sensible_fee_input(2.0),
+            base_fee_per_gas: 200,
+            ..create_l2_block(2)
+        };
+        store_custom_l2_block(&mut connection, &block2, &[]).await?;
+
+        let all_pubdata_prices = [
+            0,
+            block1.batch_fee_input.fair_pubdata_price(),
+            block2.batch_fee_input.fair_pubdata_price(),
+        ]
+        .map(U256::from);
+
+        let history = client
+            .fee_history(1_000.into(), api::BlockNumber::Latest, vec![])
+            .await?;
+        assert_eq!(history.inner.oldest_block, 0.into());
+        assert_eq!(
+            history.inner.base_fee_per_gas,
+            [0, 100, 200, 200].map(U256::from) // The latest value is duplicated
+        );
+        assert_eq!(history.l2_pubdata_price, all_pubdata_prices);
+        // Values below are not filled.
+        assert_eq!(history.inner.gas_used_ratio, [0.0; 3]);
+        assert_eq!(history.inner.base_fee_per_blob_gas, [U256::zero(); 4]);
+        assert_eq!(history.inner.blob_gas_used_ratio, [0.0; 3]);
```

Comment:
@0xVolosnikov Is duplicating the last value in `base_fee_per_gas` and not filling `gas_used_ratio`, `base_fee_per_blob_gas` and `blob_gas_used_ratio` OK for tooling we want to support?

Reply:
Yes, users may incorrectly assume that due to `gas_used_ratio` being < 0.5 the base fee for the next block should decrease. This may create problems with gas estimation, but I think that's a separate issue.

Reply:
Let's try as-is and then fix the returned value if necessary.